OpenSimRT/RealTime/tests/TestRTExtFromFile.cpp

```diff
@@ -20,7 +20,7 @@
  * @file TestRTExtFromFile.cpp
  *
  * @brief Tests the RealTimeAnalysisExtended class with data acquired from file.
- * Observed delay (with GRFMPredition) = ~30ms without SO + JR, ~42ms with
+ * Observed delay (with GRFMPredition) = ~19ms without SO + JR, ~31ms with
```

Comment:
I think the variable FPS that you observed is due to thread scheduling that can mess up with how fast the acquisition thread pushes things in the buffer. To improve this, maybe we can implement a mechanism that places things at a constant FPS since the acquisition thread has this luxury.

Reply:
Or it could just be that the SO alone ranges from 8ms to 125ms for a single frame. Pushing to the data buffer is 0ms, so this cannot be an issue. Retrieving data from the buffer has an upper cap since it cannot last any longer than the acquisition rate + IK + Filter time. 

My guess would be that the samples that the SO receives are not contiguous, and that makes it harder to find a solution close to the previous estimations. But since I did not have any issues in Windows, I could be wrong. Could it be a Ipop implementation issue? Have you tried it and experienced the same behavior?